golos.publication/golos.publication.cpp

```diff
@@ -324,11 +324,19 @@ void publication::payto(name user, eosio::asset quantity, enum_t mode, std::stri
     if(quantity.amount == 0)
         return;
 
-    if(static_cast<payment_t>(mode) == payment_t::TOKEN)
-        INLINE_ACTION_SENDER(token, payment) (config::token_name, {_self, config::active_name}, {_self, user, quantity, memo});
-    else if(static_cast<payment_t>(mode) == payment_t::VESTING)
-        INLINE_ACTION_SENDER(token, transfer) (config::token_name, {_self, config::active_name},
+    if(static_cast<payment_t>(mode) == payment_t::TOKEN) {
+        if (token::balance_exist(config::token_name, user, quantity.symbol.code())) 
+            INLINE_ACTION_SENDER(token, payment) (config::token_name, {_self, config::active_name}, {_self, user, quantity, memo});
+        else
+            INLINE_ACTION_SENDER(token, payment) (config::token_name, {user, config::active_name}, {_self, _self, quantity, memo});
```

Comment:
1. looks like this code is not tested. `do_transfer` can't transfer to self: https://github.com/GolosChain/cyberway.contracts/blob/91f4b02a17d3e66b1f0f98cb41abc472af2e5c4e/cyber.token/src/cyber.token.cpp#L123